pkg/cloud/services/eks/identity_provider.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package eks
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/eks"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/cluster-api-provider-aws/controlplane/eks/api/v1alpha4"
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/converters"
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/eks/identityprovider"
+)
+
+func (s *Service) reconcileIdentityProvider(ctx context.Context) error {
+	s.scope.Info("begin identity provider reconcile")
+	if s.scope.ControlPlane.Spec.OIDCIdentityProviderConfig == nil {
+		return nil
+	}
+
+	clusterName := s.scope.KubernetesClusterName()
+	current, err := s.getAssociatedIdentityProvider(ctx, clusterName)
+	if err != nil {
+		return errors.Wrap(err, "unable to list associated identity providers")
+	}
+
+	desired := s.convertSDKToIdentityProvider(s.scope.OIDCIdentityProviderConfig())
+
+	if desired == nil && current == nil {
+		s.scope.Info("no identity provider required or installed, no action needed")
+		return nil
+	}
+
+	s.scope.V(2).Info("got states", "desired", desired, "current", current)
```

Comment:
I think maybe we should change "got states" to something a bit more descriptive. perhaps `creating oidc provider plan`, WDYT?

Reply:
Ack